src/core/test/benchmarks/src/main/scala/Benchmarks.scala

```diff
@@ -3,98 +3,125 @@
+    assert(!newBenchmarks.map(_.name).contains(name), s"Benchmark $name already exists")
+    newBenchmarks.append(Benchmark(name, value, precision, higherIsBetter))
   }
 
-  def getFileSystem(): FileSystem = {
-    val config = sc.hadoopConfiguration
-    import org.apache.hadoop.fs.Path
-    val dfs_cwd = new Path(".")
-    dfs_cwd.getFileSystem(config)
+  def addBenchmark(name: String,
+                   value: Double,
+                   precision: Int,
+                   higherIsBetter: Boolean): Unit = {
+    addBenchmark(name, value, scala.math.pow(10, -precision.toDouble), higherIsBetter)
   }
 
-  /** Reads a CSV file given the file name and file location.
-    * @param fileName The name of the csv file.
-    * @param fileLocation The full path to the csv file.
-    * @return A dataframe from read CSV file.
-    */
-  def readCSV(fileName: String, fileLocation: String): DataFrame = {
-    session.read
-      .option("header", "true").option("inferSchema", "true")
-      .option("treatEmptyValuesAsNulls", "false")
-      .option("delimiter", if (fileName.endsWith(".csv")) "," else "\t")
-      .csv(fileLocation)
+  def addBenchmark(name: String,
+                   value: Double,
+                   precision: Int): Unit = {
+    addBenchmark(name, value, precision, higherIsBetter = true)
   }
 
-  /** Rounds the given metric to 2 decimals.
-    * @param metric The metric to round.
-    * @return The rounded metric.
-    */
-  def round(metric: Double, decimals: Int): Double = {
-    BigDecimal(metric)
-      .setScale(decimals, BigDecimal.RoundingMode.HALF_UP).toDouble
+  def addBenchmark(name: String,
+                   value: Double,
+                   precision: Double): Unit = {
+    addBenchmark(name, value, precision, higherIsBetter = true)
   }
 
-  def compareBenchmarkFiles(): Unit = {
-    try writeFile(benchmarkMetricsFile, accuracyResults.mkString("\n") + "\n")
-    catch {
-      case e: java.io.IOException => throw new Exception("Not able to process benchmarks file")
-    }
-    val historicMetrics = readFile(historicMetricsFile, _.getLines.toList)
-    if (historicMetrics.length != accuracyResults.length)
-      throw new Exception(s"Mis-matching number of lines in new benchmarks file: $benchmarkMetricsFile")
-    for (((hist,acc),i) <- (historicMetrics zip accuracyResults).zipWithIndex) {
-      assert(hist == acc,
-        s"""Lines do not match on file comparison:
-           |  $historicMetricsFile:$i:
-           |    $hist
-           |  $benchmarkMetricsFile:$i:
-           |    $acc
-           |.""".stripMargin)
-    }
+  def compareBenchmark(bmNew: Benchmark, bmOld: Benchmark): Assertion = {
+    assert(bmNew.name == bmOld.name, "Benchmark names do not match")
+    assert(bmNew.higherIsBetter == bmOld.higherIsBetter, "higherIsBetter does not match")
+    assert(bmNew.precision === bmNew.precision, "precision does not match")
+    assert(bmNew.precision >=0, "precision needs to be positive")
```

Comment:
I think precision could be negative if you are going to the 1's, 10's, hundreds etc place (by whole values)
In fact, I believe the regression benchmarks did this?
Also, the spacing is a little weird, as in the ">= 0" should be a space between op and value

Reply:
done